_posts/2024-07-08-idea-editor.adoc

```diff
@@ -0,0 +1,130 @@
+
+JetBrains https://lp.jetbrains.com/projector/[suspended] the development and support of the Projector as a standalone project and introduced a new Projector-based product - https://www.jetbrains.com/remote-development/gateway/[JetBrains Gateway]. It allows the use of the desktop IDE instead of the in-browser web client. So, we integrated its support to Eclipse Che.
+
+== How to connect desktop IntelliJ IDEA to your Che CDE?
+
+There are a few prerequisites for connecting IntelliJ IDEA to a CDE.
+
+*_Prerequisite 1: Gateway application_*
+Install the JetBrains Gateway application by following the instructions from https://www.jetbrains.com/help/idea/jetbrains-gateway.html
+
+*_Prerequisite 2: Gateway plugin_*
+Install https://plugins.jetbrains.com/plugin/24234-openshift-dev-spaces[the Gateway plugin] which provides a connector for establishing a connection between your local JetBrains IDE and your remote CDE running on the OpenShift cluster:
+
+. From the Gateway main window, open the *Manage Providers* dialog.
+. Search for the *OpenShift Dev Spaces* plugin on the *Marketplace* tab and install it.
+
+*_Prerequisite 3: OpenShift CLI_*
+
+The Gateway plugin relies on the https://docs.openshift.com/container-platform/4.16/cli_reference/openshift_cli/getting-started-cli.html[OpenShift CLI] to access an OpenShift cluster’s API server for managing CDEs.
+
+Install the OpenShift CLI following the instructions from https://docs.openshift.com/container-platform/4.15/cli_reference/openshift_cli/getting-started-cli.html#installing-openshift-cli.
+
+=== Usage scenarios
+
+JetBrains Gateway provides two main workflows:
+
+* Server-to-client flow implies a CDE is running with the specific IDE development server
+* Client-to-server flow implies the user chooses which IDE should be downloaded and running as a development server in a CDE
+
+Currently, Eclipse Che supports only server-to-client flow with the IntelliJ IDEA as a remote development server. In the future, we may add support for client-to-server flow as well.
+
+There are two options for connecting your desktop IDEA to a CDE, depending on your needs:
+
+* using the Che Dashboard
+* using the JetBrains Gateway application
+
+==== Option 1: from the Che Dashboard
+
+This option suits all needs and doesn't require you to use the Gateway application directly.
+
+It has one additional prerequisite - your local OpenShift CLI tool (oc) should be logged in to your OpenShift cluster. See https://docs.openshift.com/container-platform/4.15/cli_reference/openshift_cli/getting-started-cli.html#cli-logging-in_cli-developer-commands[this doc] for all possible ways of logging in or follow the next steps that describe the most used option:
+
+. Go to the *Applications* menu on Che Dashboard and choose the *OpenShift console*.
+. On the opened console page, go to the top-right menu and choose the *Copy login command*.
+. Copy the displayed *oc login* command and execute it in your local terminal.
```

Comment:
screenshot would be nice to have

Reply:
They are really lacking here.
Thanks for the suggestion, @ibuziuk!
I added a couple of screenshots to illustrate the steps.